app-frontend/src/app/components/lab/inputNode/inputNode.html

```diff
@@ -1,37 +1,65 @@
 <div class="input-node-item">
-  <div class="input-node-label">Project</div>
-  <div class="input-data form-group all-in-one no-margin">
-    <div class="flex-fill input-node-input-label">
-      <span ng-show="!$ctrl.selectedProject">Select a Project</span>
-      {{$ctrl.selectedProject.name | limitTo : 18}}
-      <span ng-show="$ctrl.selectedProject.name.length > 18">...</span>
+    <div class="input-node-label">Project</div>
+    <div class="input-data form-group all-in-one no-margin">
+        <div class="flex-fill input-node-input-label">
+            <span ng-show="!$ctrl.selectedProject">Select a Project</span>
+            {{$ctrl.selectedProject.name | limitTo : 18}}
+            <span ng-show="$ctrl.selectedProject.name.length > 18">...</span>
+        </div>
+        <button type="button" class="btn btn-flat" ng-click="$ctrl.selectProjectModal()">
+            Change
+        </button>
     </div>
-    <button type="button" class="btn btn-flat" ng-click="$ctrl.selectProjectModal()">Change</button>
-  </div>
 </div>
 <div class="input-node-item">
-  <div class="input-node-label">Band</div>
-  <div class="input-data form-group no-margin">
-    <rf-band-select selected-band="$ctrl.selectedBand"
-                    bands="$ctrl.bands"
-                    on-band-select="$ctrl.onBandChange(index)"
-                    disabled="!$ctrl.selectedProject"
-                    ng-if="!$ctrl.selectedProject || $ctrl.bands.length">
-    </rf-band-select>
-    <div ng-if="!$ctrl.bands.length && $ctrl.fetchingDatasources">
-      Loading band data
+    <div class="input-node-label">Layer</div>
+    <div class="input-data form-group all-in-one no-margin">
+        <div class="flex-fill input-node-input-label">
+            <span ng-show="!$ctrl.selectedLayer">Select a Layer</span> {{$ctrl.selectedLayer.name |
```

Comment:
Just a drive-by comment, but this could be done completely with the CSS `text-overflow` property https://developer.mozilla.org/en-US/docs/Web/CSS/text-overflow
The advantage would be that we can avoid hardcoding character counts. Not a big deal though.

Reply:
This is how you can use text-overflow etc to work on these elements
```diff
diff --git a/app-frontend/src/app/components/lab/inputNode/inputNode.html b/app-frontend/src/app/components/lab/inputNode/inputNode.html
index 039e0d478..ddd0a8d19 100644
--- a/app-frontend/src/app/components/lab/inputNode/inputNode.html
+++ b/app-frontend/src/app/components/lab/inputNode/inputNode.html
@@ -3,8 +3,7 @@
     <div class="input-data form-group all-in-one no-margin">
         <div class="flex-fill input-node-input-label">
             <span ng-show="!$ctrl.selectedProject">Select a Project</span>
-            {{$ctrl.selectedProject.name | limitTo : 18}}
-            <span ng-show="$ctrl.selectedProject.name.length > 18">...</span>
+            {{$ctrl.selectedProject.name}}
         </div>
         <button type="button" class="btn btn-flat" ng-click="$ctrl.selectProjectModal()">
             Change
@@ -14,9 +13,9 @@
 <div class="input-node-item">
     <div class="input-node-label">Layer</div>
     <div class="input-data form-group all-in-one no-margin">
-        <div class="flex-fill input-node-input-label">
-            <span ng-show="!$ctrl.selectedLayer">Select a Layer</span> {{$ctrl.selectedLayer.name |
-            limitTo : 18}} <span ng-show="$ctrl.selectedLayer.name.length > 18">...</span>
+        <div class="flex-fill input-node-input-label ">
+          <span ng-show="!$ctrl.selectedLayer">Select a Layer</span>
+          {{$ctrl.selectedLayer.name }}
         </div>
         <button
             type="button"
diff --git a/app-frontend/src/assets/styles/sass/_shame.scss b/app-frontend/src/assets/styles/sass/_shame.scss
index d20315873..ed64ddf83 100644
--- a/app-frontend/src/assets/styles/sass/_shame.scss
+++ b/app-frontend/src/assets/styles/sass/_shame.scss
@@ -1722,7 +1722,10 @@ rf-node-statistics {
 }
 
 .input-node-input-label {
-    display: flex;
+    text-overflow: ellipsis;
+    overflow: hidden;
+    white-space: nowrap;
+    min-width: 0;
     align-items: center;
     padding: 0.8rem 1rem;
     font-weight: 600;
```